lighthouse/tests/beacon_node.rs

```diff
@@ -0,0 +1,850 @@
+            } else {
+                panic!("Slasher config was parsed incorrectly");
+            }
+        });
+}
+#[test]
+fn slasher_max_db_size_flag() {
+    CommandLineTest::new()
+        .flag("slasher", None)
+        .flag("slasher-max-db-size", Some("10"))
+        .run()
+        .with_config(|config| {
+            let slasher_config = config
+                .slasher
+                .as_ref()
+                .expect("Unable to parse Slasher config");
+            assert_eq!(slasher_config.max_db_size_mbs, 10240);
+        });
+}
+#[test]
+fn slasher_chunk_size_flag() {
+    CommandLineTest::new()
+        .flag("slasher", None)
+        .flag("slasher-chunk-size", Some("32"))
+        .run()
+        .with_config(|config| {
+            let slasher_config = config
+                .slasher
+                .as_ref()
+                .expect("Unable to parse Slasher config");
+            assert_eq!(slasher_config.chunk_size, 32);
+        });
+}
+#[test]
+fn slasher_validator_chunk_size_flag() {
+    CommandLineTest::new()
+        .flag("slasher", None)
+        .flag("slasher-validator-chunk-size", Some("512"))
+        .run()
+        .with_config(|config| {
+            let slasher_config = config
+                .slasher
+                .as_ref()
+                .expect("Unable to parse Slasher config");
+            assert_eq!(slasher_config.validator_chunk_size, 512);
+        });
+}
+#[test]
+fn slasher_broadcast_flag() {
+    CommandLineTest::new()
+        .flag("slasher", None)
+        .flag("slasher-broadcast", None)
+        .run()
+        .with_config(|config| {
+            let slasher_config = config
+                .slasher
+                .as_ref()
+                .expect("Unable to parse Slasher config");
+            assert!(slasher_config.broadcast);
+        });
+}
+#[test]
+#[should_panic]
+fn ensure_panic_on_failed_launch() {
+    CommandLineTest::new()
+        .flag("slasher", None)
+        .flag("slasher-chunk-size", Some("10"))
+        .run()
+        .with_config(|config| {
+            let slasher_config = config
+                .slasher
+                .as_ref()
+                .expect("Unable to parse Slasher config");
+            assert_eq!(slasher_config.chunk_size, 10);
+        });
+}
+
+/// A bit of hack to find an unused port.
+///
+/// Does not guarantee that the given port is unused after the function exists, just that it was
+/// unused before the function started (i.e., it does not reserve a port).
+///
+/// Used for passing unused ports to libp2 so that lighthouse won't have to update
+/// its own ENR.
+///
+/// NOTE: It is possible that libp2p/discv5 is unable to bind to the
+/// ports returned by this function as the OS has a buffer period where
+/// it doesn't allow binding to the same port even after the socket is closed.
+/// We might have to use SO_REUSEADDR socket option from `std::net2` crate in
+/// that case.
```

Comment:
I guess we'll see how this goes on CI. I'm pinging @AgeManning to see if he has any suggestions.

Reply:
This is code I grabbed from here: https://github.com/sigp/lighthouse/blob/stable/beacon_node/src/config.rs#L639-L678
Hasn't seemed to cause any CI issues I can find. 
It's also used here: https://github.com/sigp/lighthouse/blob/stable/beacon_node/eth2_libp2p/tests/common/mod.rs#L45-L74 so I figured it was safe enough to include.

Reply:
Yep. We couldn't really find a better way of doing this, and its not used in production (unless you use like the zero port cli flag). 

I think this is fine for tests and it seems to work for the CLI, its just not 100% certain.